rotkehlchen/api/v1/schemas.py

```diff
@@ -720,49 +725,168 @@ def generate_fields_post_validation(self, data: dict[str, Any]) -> dict[str, Any
+                data['event_subtype'] = HistoryEventSubType.NONE
+            return {'event': HistoryEvent(**data)}
+
+    class CreateEthBlockEventEventSchema(BaseSchema):
+        is_mev_reward = fields.Boolean(required=True)
+        fee_recipient = EvmAddressField(required=True)
+        block_number = fields.Integer(
+            required=True,
+            validate=webargs.validate.Range(
+                min=0,
+                error='Block number must be an integer >= 0',
+            ),
+        )
+        validator_index = fields.Integer(
+            required=True,
+            validate=webargs.validate.Range(
+                min=0,
+                error='Validator index must be an integer >= 0',
+            ),
+        )
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            return {'event': EthBlockEvent(**data)}
+
+    class CreateEthDepositEventEventSchema(BaseSchema):
+        tx_hash = EVMTransactionHashField(required=True)
+        depositor = EvmAddressField(required=True)
+        event_identifier = fields.String(required=True)
+        sequence_index = fields.Integer(required=True)
+        validator_index = fields.Integer(
+            required=True,
+            validate=webargs.validate.Range(
+                min=0,
+                error='Validator index must be an integer >= 0',
+            ),
+        )
+        extra_data = fields.Dict(load_default=None)
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            return {'event': EthDepositEvent(**data)}
+
+    class CreateEthWithdrawalEventEventSchema(BaseSchema):
+        is_exit = fields.Boolean(required=True)
+        withdrawal_address = EvmAddressField(required=True)
+        validator_index = fields.Integer(
+            required=True,
+            validate=webargs.validate.Range(
+                min=0,
+                error='Validator index must be an integer >= 0',
+            ),
+        )
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            return {'event': EthWithdrawalEvent(**data)}
 
     @post_load
     def make_history_base_entry(
             self,
             data: dict[str, Any],
             **_kwargs: Any,
     ) -> dict[str, Any]:
-        if data.get('event_subtype', None) is None:
-            data['event_subtype'] = HistoryEventSubType.NONE
-
-        if data['location'] not in EVM_LOCATIONS:
-            raise ValidationError(
-                message=f'EVM event location needs to be one of {EVM_LOCATIONS}',
-                field_name='location',
-            )
+        entry_type = data.pop('entry_type')
```

Comment:
if you pop it it will be removed from the mapping. I would do a get instead. You won't use later but is something unexpected that if in the future we need to use we will find that is missing from the data

Reply:
i need to remove it from the mapping, it throws an error during validation if I don't